index.php

```diff
@@ -159,27 +266,52 @@ function create_block_theme_get_theme_css( $theme ) {
  * @param string $filename path of the ZIP file.
  */
 function gutenberg_edit_site_export_theme_create_zip( $filename, $theme ) {
-
 	$base_theme = wp_get_theme()->get('TextDomain');
 
 	if ( ! class_exists( 'ZipArchive' ) ) {
 		return new WP_Error( 'Zip Export not supported.' );
 	}
 
 	$zip = new ZipArchive();
-	$zip->open( $filename, ZipArchive::OVERWRITE );
-	$zip->addEmptyDir( $theme['slug'] );
-	$zip->addEmptyDir( $theme['slug'] . '/templates' );
-	$zip->addEmptyDir( $theme['slug'] . '/parts' );
+	$zip->open( $filename, ZipArchive::CREATE | ZipArchive::OVERWRITE );
+
+	// If this is a standalone theme then copy the whole theme into a zip folder.
+	if ( $theme['type'] === "block" ) {
+		// Get real path for our folder
+		$theme_path = get_stylesheet_directory();
+
+		// Create recursive directory iterator
+		/** @var SplFileInfo[] $files */
+		$files = new RecursiveIteratorIterator(
+			new RecursiveDirectoryIterator( $theme_path ),
+			RecursiveIteratorIterator::LEAVES_ONLY
+		);
+
+		foreach ( $files as $name => $file )
+		{
+			// Skip directories (they would be added automatically)
+			if ( ! $file->isDir() )
+			{
+				// Get real and relative path for current file
+				$file_path = $file->getRealPath();
+				$relative_path = substr( $file_path, strlen( $theme_path ) + 1 );
+
+				// Add current file to archive
+				$zip->addFile( $file_path, $relative_path );
+			}
+		}
+	}
 
 	// Load templates into the zip file.
 	$templates = gutenberg_get_block_templates();
+	if ( $templates ) {
+		$zip->addEmptyDir( 'templates' );
+	}
```

Comment:
Exporting a "standalone" theme where the current them uses the out-of-date templates directory results in the exported theme having templates in BOTH locations.  

I think better would be to have this exported location exclusively (no matter where the original has them).  It's best to export them (to pick up the user changes).

Alternately we could just choose to export them where the current theme has them declared.

Reply:
I've added code to exclude the templates during "copy all the things".  I think this addresses things here.